src/types/types.ts

```diff
@@ -154,11 +154,21 @@ export type SearchRequestGET = Pagination &
     retrieveVectors?: boolean;
   };
 
+export type FederationOptions = { weight: number };
+export type MultiSearchFederation = { limit?: number; offset?: number };
+
 export type MultiSearchQuery = SearchParams & { indexUid: string };
+export type MultiSearchQueryWithFederation = MultiSearchQuery & {
+  federationOptions?: FederationOptions;
+};
 
 export type MultiSearchParams = {
   queries: MultiSearchQuery[];
 };
+export type FederatedMultiSearchParams = {
+  federation: MultiSearchFederation;
+  queries: MultiSearchQueryWithFederation[];
+};
```

Comment:
I'm not a big fan of those, because of the repetitions. 

I think we can find a more elegant way of typing this, so what about typing the `MultiSearchResponse` differently to accept those changes?

I tried locally with the following: 
```
export type MultiSearchResponse<T = Record<string, any>> = {
  results: Array<MultiSearchResult<T>>;
  hits: Array<
    Hit<T> & {
      _federation: {
        indexUid: string;
        queriesPosition: number;
        weightedRankingScore: number;
      };
    }
  >;
};
``` 

And then you can replace those with: 
```
export type MultiSearchQuery = SearchParams & {
  indexUid: string;
  federationOptions?: {
    weight?: number;
  };
};

export type MultiSearchParams = {
  queries: MultiSearchQuery[];
  federation?: {
    limit?: number;
    offset?: number;
  };
};
```

I don't know if there are potential mistakes in it (I am no typescript expert), so I'd love you hear what you think about it :)

Reply:
About typing `MultiSearchResponse` with `hits` that contain `_federation`, we haven't done it for `_formatted`, `_matchesPosition` and the lot, it would be inconsistent doing it now just for `_federation`.
I think it's better to leave it as is and address this type improvement in a separate issue, and then start doing it for all of them where possible.

I think I am starting to see what you're trying to do. This way we would be providing false types to the users. My implementation conditionally gives you the right type depending on the shape of `queries`, your way sacrifices all that for simplicity. But I'd argue type correctness should come first.

Reply:
Ok got you 👍  Let's keep it as is then 🙌